tests/python_compiler/conftest.py

```diff
@@ -43,7 +61,48 @@ def _run_filecheck(program_str, xdsl_module):
 
 @pytest.fixture(scope="function")
 def run_filecheck():
+    """Fixture to run filecheck on an xDSL module."""
     if not filecheck_available:
         pytest.skip("Cannot run lit tests without filecheck.")
 
-    yield _run_filecheck
+    yield _run_filecheck_impl
+
+
+def _from_qjit_impl(qjit_fn):
+    """Create a wrapper around a QJIT-ed function that returns an xDSL module."""
+
+    def wrapper(*args, **kwargs):
+        _ = qjit_fn(*args, **kwargs)
+        mod = qjit_fn.mlir
+        mod_generic = _quantum_opt(
+            _quantum_opt(
+                ("--pass-pipeline", "builtin.module(canonicalize)"),
+                "-mlir-print-op-generic",
+                stdin=mod,
+            )
+        )
+
+        ctx = Context(allow_unregistered=True)
+        ctx.load_dialect(arith.Arith)
+        ctx.load_dialect(builtin.Builtin)
+        ctx.load_dialect(func.Func)
+        ctx.load_dialect(linalg.Linalg)
+        ctx.load_dialect(scf.Scf)
+        ctx.load_dialect(stablehlo.StableHLO)
+        ctx.load_dialect(tensor.Tensor)
+        ctx.load_dialect(transform.Transform)
+        ctx.load_dialect(Quantum)
+
+        xmod = Parser(ctx, mod_generic).parse_module()
+        return xmod
+
+    return wrapper
+
+
+@pytest.fixture(scope="function")
+def from_qjit():
```

Comment:
Instead of a fixture, this could potentially be part of the `qml.compiler.python_compiler` submodule.